server/test/services/withings/lib/withings.init.test.js

```diff
@@ -0,0 +1,358 @@
+const { assert } = require('chai');
+const { OAuth2Server } = require('oauth2-mock-server');
+const { fake } = require('sinon');
+const ServerMock = require('mock-http-server');
+const OAuth2Manager = require('../../../../lib/oauth2/index');
+const WithingsHandler = require('../../../../services/withings/lib');
+const { OAUTH2 } = require('../../../../utils/constants.js');
+
+const serverOauth2 = new OAuth2Server();
+const server = new ServerMock({ host: 'localhost', port: 9192 }, null);
+
+const gladys = {
+  device: {
+    create: fake.returns(null),
+    destroyByServiceId: fake.returns(null),
+    get: function getFakeDevice() {
+      return [
+        {
+          name: 'Withings - string',
+          params: [
+            {
+              name: 'WITHINGS_DEVICE_ID',
+              value: 'withingsDevideId4',
+            },
+          ],
+        },
+      ];
+    },
+    saveHistoricalState: function shs(device, featureBattery, featureState) {
+      device.featureBattery = featureBattery;
+      device.featureBattery = featureState;
+    },
+  },
+  user: {
+    get: fake.returns([{ id: '0cd30aef-9c4e-4a23-88e3-3547971296e5' }]),
+  },
+  event: { emit: fake.returns(null) },
+  variable: {
+    getValue: function returnValue(key, serviceId) {
+      switch (key) {
+        case `${OAUTH2.VARIABLE.TOKEN_HOST}`:
+          return 'http://localhost:9292';
+        case `${OAUTH2.VARIABLE.TOKEN_PATH}`:
+          return '/oauth2/token';
+        case OAUTH2.VARIABLE.REDIRECT_URI_SUFFIX:
+          return 'ashboard/integration/health/test/settings';
+        case `${OAUTH2.VARIABLE.AUTHORIZE_HOST}`:
+          return 'http://localhost:9292';
+        case `${OAUTH2.VARIABLE.AUTHORIZE_PATH}`:
+          return '/oauth2_user/authorize2';
+        case `${OAUTH2.VARIABLE.GRANT_TYPE}`:
+          return 'authorization_code';
+        case `${OAUTH2.VARIABLE.INTEGRATION_SCOPE}`:
```

Comment:
uncesseray string concat.

Maybe you can do a search on `${` to find all of these?

Reply:
i fix it